### DISCUSSION TITLE
Bypass list for Ruleset not applying to status checks and pull requests

### DISCUSSION THREAD

**Discussion by aiell0** [community/community#86534] [Repositories]
### Select Topic Area

Bug

### Body

I have a repository with the default branch of `main`. I have created a ruleset to protect the main branch. In this ruleset, I have configured the following:

* Require a PR before merging
  * Also require a code review from a codeowner
* Require status checks to pass
  * I have 3 status checks listed here

There is a github user in our organization that is strictly for automation purposes only, that I would like to allow to push to the main branch and bypass all of these rules. I have placed said user in their own team in our organization, and then placed that team inside the bypass list for the ruleset.

However, during the rule evaluations, it is showing up as failing the two checks I have listed above despite being in the bypass list. Is there something I am missing? My understanding is that the bypass list allows specific teams to ignore the rulesets for that branch.

**Comment by knownothing10** [comment]
🤔 It seems like you've set up a ruleset to protect the main branch in your GitHub repository and added a user for automation purposes. You've placed that user in a team within your organization and added that team to the bypass list of the ruleset. However, you're experiencing issues where the user is still failing the checks despite being in the bypass list.

There are a few possible reasons why this might be happening. Let's go through them:

Double-check the branch: Make sure you've correctly configured the ruleset for the main branch. Ensure that you haven't accidentally set up the ruleset for a different branch.

Verify team membership: Confirm that the automation user is indeed a member of the team added to the bypass list. Check the team's membership and ensure the user is included.

Confirm ruleset configuration: Revisit the ruleset configuration and verify that all settings are accurate. Ensure that the bypass list includes the correct team and that there are no conflicting or overriding rules that could be affecting the automation user.

Check organization permissions: Ensure that the team the automation user is part of has the necessary permissions within the organization and repository. They should have the required access to bypass the ruleset.

Review the order of rules: If you have other rulesets or checks that are evaluated before the one in question, they might override the bypass list. Make sure the ruleset you're working with is properly ordered in relation to other rulesets or checks.

If you've reviewed the above points and are still facing issues, you might want to consider reaching out to GitHub support or consulting the GitHub documentation for further assistance.

**Reply by dbeilin** [reply]
Thanks ChatGPT 👍

**Comment by mkarnicki** [comment]
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/creating-rulesets-for-a-repository#granting-bypass-permissions-for-your-ruleset

> The selected actor is now required to open a pull request to make changes to a repository, creating a clear trail of their changes in the pull request and audit log. The actor can then choose to bypass any branch protections and merge that pull request.

So the checks may fail, but the user added in the team added to the bypass ruleset list is able to still merge the PR.

**Comment by navrkald** [comment]
Hi @aiell0, if you have configured your bypass list to be `Allowed always` then pushing to your protected branch should work normally without any problems. We faced same issue, but it turned out we wasn't correctly passing our GH App credentials to checkout and push steps.

@mkarnicki your Github documentation citation is for case when you select in bypass list `For pull request only`.
